unit_tests/src/test_irods_lifetime_manager.cpp

```diff
@@ -0,0 +1,143 @@
+#include "catch.hpp"
+
+#include "lifetime_manager.hpp"
+#include "objInfo.h"
+#include "rcMisc.h"
+
+namespace {
+    template<typename T>
+    T* get_pointer_to_struct(const int num = 1)
+    {
+        const size_t size = num * sizeof(T);
+        T* t = (T*)malloc(size);
+        std::memset(t, 0, size);
+        return t;
+    }
+
+    class test_class
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class 'test_class' does not have a constructor although it has private member variables.](https://app.codacy.com/app/trel/irods/pullRequest?prid=4508572)